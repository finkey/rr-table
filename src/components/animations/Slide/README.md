## Slide

Modals that appears from the right of the screen.
