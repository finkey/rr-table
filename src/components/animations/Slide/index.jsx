import React from 'react';
import PropTypes from 'prop-types';
import Transition from 'react-transition-group/Transition';
import styled from 'styled-components';

const Wrapper = styled.div`
  height: 100%;
  transition: all 300ms ease-in-out;

  visibility: ${({ status }) => {
    switch (status) {
      case 'entering':
        return 'hidden';
      case 'entered':
        return 'visible';
      case 'exiting':
        return 'visible';
      case 'exited':
        return 'hidden';
      default:
        return null;
    }
  }};

  opacity: ${({ status }) => {
    switch (status) {
      case 'entering':
        return 0;
      case 'entered':
        return 1;
      case 'exiting':
        return 1;
      case 'exited':
        return 0;
      default:
        return null;
    }
  }};

  transform: translate3d(
    ${({ status }) => {
    let tx = '100%';
    switch (status) {
      case 'entering':
        tx = 0;
        break;
      case 'entered':
        tx = 0;
        break;
      case 'exiting':
        tx = '100%';
        break;
      case 'exited':
        tx = '100%';
        break;
      default:
        tx = '100%';
    }
    return `${tx}, 0, 0`;
  }}
  );
`;

const Slide = ({ children, isOpen, delay }) => (
  <Transition
    in={isOpen}
    timeout={{
      enter: 0,
      exit: delay,
    }}
  >
    {status => <Wrapper status={status}>{children}</Wrapper>}
  </Transition>
);

Slide.propTypes = {
  children: PropTypes.node,
  isOpen: PropTypes.bool,
  delay: PropTypes.number,
};

Slide.defaultProps = {
  delay: 300,
};

export default Slide;
