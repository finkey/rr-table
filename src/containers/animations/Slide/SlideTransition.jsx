import React from 'react';
import Transition from 'react-transition-group/Transition';

const SlideTransition = ({ children, isOpen, delay = 300 }) => (
  <Transition
    in={isOpen}
    timeout={{
      enter: 0,
      exit: delay,
    }}
  >
    {status => children(status)}
  </Transition>
);

export default SlideTransition;
