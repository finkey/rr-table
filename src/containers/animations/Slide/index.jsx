import React from 'react';
import SlideTransition from './SlideTransition';
import SlideComp from './SlideComp';

const Slide = ({ children, isOpen, delay }) => (
  <SlideTransition isOpen={isOpen} delay={delay}>
    {status => <SlideComp status={status}>{children}</SlideComp>}
  </SlideTransition>
);

export default Slide;
