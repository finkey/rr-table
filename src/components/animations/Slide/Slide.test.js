import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Slide from './index';

describe('Slide', () => {
  test('should match snapshot', () => {
    const comp = renderer.create(<Slide />);
    const tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
