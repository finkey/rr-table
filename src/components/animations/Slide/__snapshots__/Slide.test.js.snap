// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should match snapshot 1`] = `
.c0 {
  height: 100%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  visibility: hidden;
  opacity: 0;
  -webkit-transform: translate3d( 100%,0,0 );
  -ms-transform: translate3d( 100%,0,0 );
  transform: translate3d( 100%,0,0 );
}

<div
  className="c0"
/>
`;
